api/clever/__init__.py

```diff
@@ -224,59 +250,57 @@ def remote_patron_lookup(self, token):
         result = self._get(self.CLEVER_API_BASE_URL + '/me', bearer_headers)
         data = result.get('data', {}) or {}
 
         identifier = data.get('id', None)
 
         if not identifier:
-            return INVALID_CREDENTIALS.detailed(
-                _("A valid Clever login is required.")
-            )
+            return INVALID_CREDENTIALS.detailed(lgt("A valid Clever login is required."))
 
         if result.get('type') not in self.SUPPORTED_USER_TYPES:
             return UNSUPPORTED_CLEVER_USER_TYPE
 
         links = result['links']
 
-        user_link = [l for l in links if l['rel'] == 'canonical'][0]['uri']
+        user_link = [link for link in links if link['rel'] == 'canonical'][0]['uri']
+        # The canonical link includes the API version, so we use the base URL.
         user = self._get(self.CLEVER_API_BASE_URL + user_link, bearer_headers)
 
         user_data = user['data']
         school_id = user_data['school']
-        school = self._get(
-            self.CLEVER_API_BASE_URL + '/v1.1/schools/%s' % school_id,
-            bearer_headers
-        )
-
+        school = self._get(f"{self.CLEVER_API_BASE_URL}/v1.1/schools/{school_id}", bearer_headers)
         school_nces_id = school['data'].get('nces_id')
 
         # TODO: check student free and reduced lunch status as well
 
         if school_nces_id is None:
-            self.log.error(
-                "No NCES ID found in Clever school data: %s", repr(school)
-            )
+            self.log.error("No NCES ID found in Clever school data: %s", repr(school))
             return CLEVER_UNKNOWN_SCHOOL
 
         if school_nces_id not in TITLE_I_NCES_IDS:
             self.log.info("%s didn't match a Title I NCES ID", school_nces_id)
             return CLEVER_NOT_ELIGIBLE
 
+        external_type = None
+
         if result['type'] == 'student':
-            grade = user_data.get('grade')
-            external_type = None
-            if grade in ["Kindergarten", "1", "2", "3"]:
-                external_type = "E"
-            elif grade in ["4", "5", "6", "7", "8"]:
-                external_type = "M"
-            elif grade in ["9", "10", "11", "12"]:
-                external_type = "H"
+            # We need to be able to assign an external_type to students, so that they
+            # get the correct content level. To do so we rely on the grade field in the
+            # user data we get back from Clever. Their API doesn't guarantee that the
+            # grade field is present, so we supply a default.
+            student_grade = user_data.get('grade', None)
+
+            if not student_grade:   # If no grade was supplied, log the school/student
+                msg = (f"CLEVER_UNKNOWN_PATRON_GRADE: School with NCES ID {school_nces_id} "
+                       f"did not supply grade for student {user_data.get('id')}")
+                self.log.info(msg)
+
+            # If we can't determine a type from the grade level, set to "A"
```

Comment:
This comment seems to go with the other branch of the `if`, which already has a comment.